tests/bsearch-different-types/repro.cc

```diff
@@ -4,9 +4,17 @@
 bool checkForImplicitGCCall(const char *name) {
   static const std::string names[] = {
     "A",
+#ifdef INVALID
+    "C",
+    "B",
+#else
     "B",
     "C",
+#endif
   };
-  return binary_search(&names[0], &names[sizeof(names) / sizeof(std::string)], name);
+  return std::binary_search(&names[0], &names[sizeof(names) / sizeof(std::string)], name);
```

Comment:
```suggestion
  return std::binary_search(std::begin(names), std::end(names), name);
```

Reply:
This code is coming from OSS so I'd rather keep it as is.